R/module-repository.R

```diff
@@ -411,10 +411,48 @@ setMethod(
           tmpFile <- file.path(tempdir(), "SpaDES_module_data") %>%
             checkPath(create = TRUE) %>%
             file.path(., xFile)
-          message("Downloading ", chksums$actualFile[id], " for module ", module, " ...")
-          download.file(x, destfile = tmpFile, mode = "wb", quiet = quiet)
-          copied <- file.copy(from = tmpFile, to = destfile, overwrite = TRUE)
-          destfile
+
+          if(interactive()) {
+            if(inherits(
+              try(suppressWarnings(
+                download.file(x, destfile = tmpFile, mode = "wb", quiet = quiet)
+              ),
+              silent = TRUE), "try-error")) {
+              ## if the URL doesn't work allow the user to retrieve it manually
+              message("Cannot download ", xFile, " for module ", module, " ...",
+                      "\nCannot open URL '", x, "'")
+              readline(prompt = paste0("Try downloading this file manually and put it in ",
+                                       module, "/data.\nPress [enter] to continue"))
+              ## re-checksums
+              chksums <- checksums(module, path) %>%
+                mutate(renamed = NA, module = module)
+            } else {
+              message("Downloading ", chksums$actualFile[id], " for module ", module, " ...")
+              download.file(x, destfile = tmpFile, mode = "wb", quiet = quiet)
+              copied <- file.copy(from = tmpFile, to = destfile, overwrite = TRUE)
+              destfile
+            }
+          } else {
```

Comment:
why is this section duplicated?

Reply:
because I thought it'd be nice for R to still do something even if it's not on interactive mode. In this case, the warning message will still be returned (and checksums repeated) if the download fails.